File: pocs/code-review-demo/agents/dev_agent.py
from beeai_framework.agents.experimental import RequirementAgent
from beeai_framework.agents.experimental.requirements.conditional import ConditionalRequirement
from beeai_framework.memory import UnconstrainedMemory
from beeai_framework.tools.think import ThinkTool
from core.backend import get_llm


def create_dev_agent() -> RequirementAgent:
    return RequirementAgent(
        llm=get_llm(),
        memory=UnconstrainedMemory(),
        tools=[ThinkTool()],
        requirements=[
            ConditionalRequirement(ThinkTool, force_at_step=1),
        ],
        role="dev_reviewer",
        instructions=(
            "You are a senior software engineer reviewing code quality. "
            "Focus on: naming conventions, code patterns, readability, and duplication. "
            "Think carefully before responding. Report each issue with a severity: info, warning, or critical."
        ),
    )

File: pocs/code-review-demo-ts/src/agents/devAgent.ts
import { RequirementAgent } from "beeai-framework/agents/requirement/agent";
import { ConditionalRequirement } from "beeai-framework/agents/requirement/requirements/conditional";
import { ThinkTool } from "beeai-framework/tools/think";

import { getLLM } from "../core/backend.js";

export async function createDevAgent() {
  return new RequirementAgent({
    llm: await getLLM(),
    tools: [new ThinkTool()],
    instructions:
      `You are a senior developer. Review the submitted code for naming conventions, readability, code patterns, and duplication. 
      Return a structured list of findings with severity: CRITICAL, WARNING, or INFO.`,
    requirements: [
      new ConditionalRequirement(ThinkTool, { forceAtStep: 1, maxInvocations: 3 }),
    ],
  });
}
